react/components/FacetCheckboxList.js

```diff
@@ -1,37 +1,59 @@
 import classNames from 'classnames'
-import React from 'react'
+import React, { useState, useContext } from 'react'
 import { Checkbox } from 'vtex.styleguide'
 
 import useSelectedFilters from '../hooks/useSelectedFilters'
 import styles from '../searchResult.css'
+import { useSortFacets } from '../constants/SearchHelpers'
+import SettingsContext from './SettingsContext'
+import SearchFilterBar from './SearchFilterBar'
 
-const FacetCheckboxList = ({ facets, onFilterCheck }) => {
+const FacetCheckboxList = ({ facets, onFilterCheck, title }) => {
   const facetsWithSelected = useSelectedFilters(facets)
+  const { orderFacetsBy, showFacetSearch, showFacetQuantity } = useContext(
+    SettingsContext
+  )
+  const [searchTerm, setSearchTerm] = useState('')
+  const sortFunc = useSortFacets(orderFacetsBy)
 
-  return facetsWithSelected.map(facet => {
-    const { name } = facet
+  return (
+    <>
+      {showFacetSearch ? (
+        <SearchFilterBar title={title} handleChange={setSearchTerm} />
+      ) : null}
+      {facetsWithSelected
+        .filter(
```

Comment:
The code looks nice but I think this may be an unnecessary burden on low-end devices.
I would apply `filter` and `sort` only if necessary:
`filter` if a searchTerm is present
`sort` if the is a value for orderFacetsBy that makes a sort needed

Reply:
Also, will this code re-execute every time there is a rerender? Can you optimize this, please? Maybe with an `useMemo` or something similar.